From 29f27f4eb83a47de85c72d4ab1663a7c3cedbac3 Mon Sep 17 00:00:00 2001
From: Cyber Knight <cyberknight755@gmail.com>
Date: Wed, 24 Jan 2024 21:58:40 +0800
Subject: [PATCH 05/16] Settings: Add a controller for gestures control

- This is done to hide offscreen gestures page when KeyHandler is disabled.

Change-Id: I26d756d045f531757f70ac311609ea201be20436
Signed-off-by: Cyber Knight <cyberknight755@gmail.com>
---
 res/xml/gestures.xml                          |  3 +-
 .../OffscreenGestureSettingsController.java   | 33 +++++++++++++++++++
 2 files changed, 35 insertions(+), 1 deletion(-)
 create mode 100644 src/com/android/settings/gestures/OffscreenGestureSettingsController.java

diff --git a/res/xml/gestures.xml b/res/xml/gestures.xml
index 2325362..42023f5 100644
--- a/res/xml/gestures.xml
+++ b/res/xml/gestures.xml
@@ -91,6 +91,7 @@
         android:key="offscreen_gesture"
         android:title="@string/offscreen_gesture_title"
         android:summary="@string/offscreen_gesture_summary"
-        android:fragment="com.android.settings.gestures.OffscreenGestureSettings" />
+        android:fragment="com.android.settings.gestures.OffscreenGestureSettings"
+        settings:controller="com.android.settings.gestures.OffscreenGestureSettingsController" />
 
 </PreferenceScreen>
diff --git a/src/com/android/settings/gestures/OffscreenGestureSettingsController.java b/src/com/android/settings/gestures/OffscreenGestureSettingsController.java
new file mode 100644
index 0000000..0b7e97f
--- /dev/null
+++ b/src/com/android/settings/gestures/OffscreenGestureSettingsController.java
@@ -0,0 +1,33 @@
+/*
+ * Copyright (C) 2024 Paranoid Android
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.android.settings.gestures;
+
+import android.content.Context;
+import com.android.internal.R;
+import com.android.settings.core.BasePreferenceController;
+
+public class OffscreenGestureSettingsController extends BasePreferenceController {
+
+    public OffscreenGestureSettingsController(Context context, String preferenceKey) {
+        super(context, preferenceKey);
+    }
+
+    @Override
+    public int getAvailabilityStatus() {
+        // Disable offscreen gestures page if KeyHandler is disabled.
+        final boolean mKeyHandlerEnabled = mContext.getResources().getBoolean(R.bool.config_enableKeyHandler);
+        return mKeyHandlerEnabled ? AVAILABLE : UNSUPPORTED_ON_DEVICE;
+    }
+}
-- 
2.47.1

